File: src/main/java/com/vaadin/demo/component/stepper/StepperStates.java
package com.vaadin.demo.component.stepper;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.stepper.Step;
import com.vaadin.flow.component.stepper.Stepper;
import com.vaadin.flow.router.Route;

@Route("stepper-states")
public class StepperStates extends Div {

    public StepperStates() {
        // tag::snippet[]
        Stepper stepper = new Stepper();
        
        Step completed = new Step("Completed step");
        completed.setState(Step.State.COMPLETED);
        
        Step active = new Step("Active step");
        active.setState(Step.State.ACTIVE);
        
        Step error = new Step("Error step");
        error.setState(Step.State.ERROR);
        
        Step inactive = new Step("Inactive step");
        
        stepper.add(completed, active, error, inactive);
        // end::snippet[]

        add(stepper);
    }

    public static class Exporter extends DemoExporter<StepperStates> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/stepper/stepper-states.ts
import '@vaadin/stepper';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('stepper-states')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-stepper>
        <vaadin-step label="Completed step" state="completed"></vaadin-step>
        <vaadin-step label="Active step" state="active"></vaadin-step>
        <vaadin-step label="Error step" state="error"></vaadin-step>
        <vaadin-step label="Inactive step"></vaadin-step>
      </vaadin-stepper>
      <!-- end::snippet[] -->
    `;
  }
}
